core/src/com/google/inject/internal/InternalFlags.java

```diff
@@ -53,10 +53,25 @@
    * The options for Guice custom class loading.
    */
   public enum CustomClassLoadingOption {
-    /** No custom class loading */
+    /** Never create child class loaders, use Unsafe to define types in existing class loaders. */
     OFF,
-    /** Automatically bridge between class loaders (Default) */
-    BRIDGE
+
+    /**
+     * Like OFF but uses Unsafe.defineAnonymousClass to reduce the cost of fast/enhanced classes.
+     *
+     * <p>Note: when using this option you cannot look up enhanced types by name or mock/spy them.
+     */
+    ANONYMOUS,
+
+    /** Prefer Unsafe, but fall back to child class loaders if Unsafe is not available (Default) */
```

Comment:
can we clarify how "Prefer Unsafe" differs from "Unsafe.defineAnonymousClass"?  what Unsafe method is it using?  (or is it defineAnonymousClass?)

same in OFF's docs

Reply:
For the general `Unsafe` case we use `Unsafe.defineAnonymousClass` to get access to the host classloader's `defineClass` method. Once we have access we then define classes using that classloader. We only use `defineAnonymousClass` to define all classes when `guice_custom_class_loading` is set to `ANONYMOUS`.

I'll expand on this in the javadoc.

Reply:
6f29cde